lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1881,9 +1876,58 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
         break;
       }
       case SymbolicValue::Array: {
-        // FIXME: Handle array attributes.
+        CanType elementType;
+        auto rawElements = attrValue.getArrayValue(elementType);
+        SmallVector<SymbolicValue, 4> elements;
+        elements.reserve(rawElements.size());
+        for (auto elt : rawElements)
+          elements.push_back(elt.lookThroughSingleElementAggregates());
+
+        auto elementTypeString = elementType->getString();
+
+        // TODO: TF_SetAttrTypeList
+
+        if (elementTypeString == "String") {
+          SmallVector<const void*, 4> pointers;
+          SmallVector<size_t, 4> sizes;
+          for (auto elt : elements) {
+            auto bytes = elt.getStringValue();
+            pointers.push_back(bytes.data());
+            sizes.push_back(bytes.size());
+          }
+          TF_SetAttrStringList(op, name.c_str(), pointers.data(), sizes.data(),
+                               elements.size());
+          break;
+        }
+        if (StringRef(elementTypeString).startswith("Int")) {
+          SmallVector<int64_t, 4> values;
+          for (auto elt : elements)
+            values.push_back(elt.getIntegerValue().getLimitedValue());
+          TF_SetAttrIntList(op, name.c_str(), values.data(), values.size());
+          break;
+        }
+        if (elementTypeString == "Float" || elementTypeString == "Double") {
+          SmallVector<float, 4> values;
+          for (auto elt : elements) {
+            auto value = elt.getFloatValue();
+            bool losesInfo = false;
```

Comment:
add a comment on why losing precision here is ok?

Reply:
well it really isn't, but this is tensorflow.... :-)